src/nimony/sem.nim

```diff
@@ -222,15 +222,28 @@ proc semInclude(c: var SemContext; it: var Item) =
 
   producesVoid c, info, it.typ
 
-proc importSingleFile(c: var SemContext; f1, origin: string; info: PackedLineInfo) =
-  let f2 = resolveFile(c, origin, f1)
+proc importSingleFile(c: var SemContext; f1: ImportedFilename; origin: string; info: PackedLineInfo) =
+  let f2 = resolveFile(c, origin, f1.path)
   let suffix = moduleSuffix(f2, c.g.config.paths)
   if not c.processedModules.containsOrIncl(suffix):
     c.meta.importedFiles.add f2
     if needsRecompile(f2, suffix):
       selfExec c, f2
 
-    loadInterface suffix, c.importTab
+    let moduleName = pool.strings.getOrIncl(f1.name)
+    let moduleSym = identToSym(c, moduleName, ModuleY)
+    let s = Sym(kind: ModuleY, name: moduleSym, pos: ImportedPos)
+    c.currentScope.addOverloadable(moduleName, s)
+    var moduleDecl = createTokenBuf(2)
+    moduleDecl.addParLe(ModuleY, info)
+    moduleDecl.addParRi()
+    publish moduleSym, moduleDecl
+    var module = ImportedModule()
+    loadInterface suffix, module.iface
+    # merge module symbols into import table:
+    for strId, symIds in module.iface:
+      c.importTab.mgetOrPut(strId, @[]).add(symIds)
```

Comment:
The code was carefully crafted to avoid things like these. Can it be avoided?

Reply:
Maybe `importTab` can change to a mapping of identifiers to a list of modules that contain that identifier, then instead of iterating over `importTab` for the symbols directly, we iterate over the modules then over their symbols with the same name as the identifier. Would this avoid problems?

Reply:
Fine, and that's how my other compiler did it.